### DISCUSSION TITLE
Removing random stuffs covering the IEDs

### DISCUSSION THREAD

**Discussion by XDSenDX** [Vdauphin/HeartsAndMinds#1561] [Question and answer] [answered]
I remember this being mentioned somewhere but i can't recall it. I'm trying to implement IEDD ied system which already has random stuffs around it, would be good if this is possible.

Cheers.

**Comment by XDSenDX** [comment]
You can join the IEDD discord right here https://discord.gg/AbdmyE2bby

**Reply by XDSenDX** [reply]
Anyways the IEDD dev helped me with the IED implementation and it seems to be a success. However some H&M junk are not letting me defuse the IED.

**Reply by Vdauphin** [reply]
What is your problem?
Please elaborate.

**Reply by XDSenDX** [reply]
The wreck are blocking my view from the IED as the IED are below them. Sometimes ieds blow up themselves because they are collided with the wreck/ junks.

Cheers.

**Reply by Vdauphin** [reply, marked as answer]
> The wreck are blocking my view from the IED as the IED are below them.

IED spawn randomly in a circle of 2 m https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/ied/create.sqf#L49C16-L49C26
This makes some IED not easy or impossible to disarm. This is intended to force player to avoid the area and change his plan or use high caliber to clear the road

> Sometimes ieds blow up themselves because they are collided with the wreck/ junks.

This can't happen has only vehicles with player side can trigger the IED: 
https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/ied/checkLoop.sqf#L34
https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/ied/checkLoop.sqf#L49

Wreck/junks are simple objects: https://community.bistudio.com/wiki/Arma_3:_Simple_Objects (https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/fnc/ied/create.sqf#L42), there are no part of `allvehicles`

**Reply by XDSenDX** [reply]
Thanks for pointing everything out, i will check the ied exploding on it own again.


Cheers and thanks for your time.

**Reply by XDSenDX** [reply]
Is there anyway to tweak what heights does the ied spawn?

**Reply by Vdauphin** [reply]
Yes: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L254-L259

**Reply by XDSenDX** [reply]
> Yes: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L254-L259

Thanks.

**Comment by Vdauphin** [comment]
> I remember this being mentioned somewhere but i can't recall it.

https://github.com/Vdauphin/HeartsAndMinds/discussions/1554

**Comment by OverlordZorn** [comment]
> This can't happen has only vehicles with player side can trigger the IED

does a player infantry man also count as player in vehicles? We had a lot of IED's going off when players just walked near the IED.

Or is this a recent change?

**Reply by XDSenDX** [reply]
> > This can't happen has only vehicles with player side can trigger the IED
> 
> does a player infantry man also count as player in vehicles? We had a lot of IED's going off when players just walked near the IED.
> 
> Or is this a recent change?

Idk but my problem is when a player walks into a city zone i think the ied got flung in the air and exploded. Or something set it off even the player is miles away from it. My other theory is collision.

**Reply by Vdauphin** [reply]
> does a player infantry man also count as player in vehicles? We had a lot of IED's going off when players just walked near the IED.

Well it is the point, players are child to `allVehicles`, see: http://vdauphin.github.io/HeartsAndMinds/InGame-documentation#ied

> Idk but my problem is when a player walks into a city zone i think the ied got flung in the air and exploded.

No there are not, what you describe is most likely car accident. AI run into each other

**Reply by Vdauphin** [reply]
> Or is this a recent change?

Nothing changed

**Reply by XDSenDX** [reply]
> > does a player infantry man also count as player in vehicles? We had a lot of IED's going off when players just walked near the IED.
> 
> Well it is the point, players are child to `allVehicles`, see: http://vdauphin.github.io/HeartsAndMinds/InGame-documentation#ied
> 
> > Idk but my problem is when a player walks into a city zone i think the ied got flung in the air and exploded.
> 
> No there are not, what you describe is most likely car accident. AI run into each other

Got it.
